wlauto/common/android/device.py

```diff
@@ -670,6 +670,19 @@ def get_device_model(self):
         except KeyError:
             return None
 
+    def is_network_connected(self):
+            network_host = 'www.google.com'
+            self.logger.debug('Checking for internet connectivity.')
+            output = adb_shell(self.adb_name, 'ping -q -w 1 -c 1 {}'.format(network_host),
```

Comment:
I believe this is present on all android devices and is part of the standard binaries included. It is part of busybox anyway (which I believe is a requirement for WA?) so it should be in the path anyway.